src/transformers/cache_utils.py

```diff
@@ -824,8 +824,8 @@ def __init__(self, config: PretrainedConfig, max_batch_size: int, max_cache_len:
             # breaks when updating the cache.
             new_layer_key_cache = torch.zeros(cache_shape, dtype=self.dtype, device=device)
             new_layer_value_cache = torch.zeros(cache_shape, dtype=self.dtype, device=device)
-            torch._dynamo.mark_static_address(new_layer_key_cache)
-            torch._dynamo.mark_static_address(new_layer_value_cache)
+            # torch._dynamo.mark_static_address(new_layer_key_cache)
+            # torch._dynamo.mark_static_address(new_layer_value_cache)
```

Comment:
This is in `__init__` function. A proper fix would be to hoist the construction of the StaticCache such that it is outside of `torch.compile` scope.

Reply:
We can maybe compile _generate, and generate inits the cache then call the compiled _generate.

Reply:
Ok, I looked more into why this was not a problem for `gpt-fast`.

In `gpt-fast`, the KV cache is an instance of nn.Module, with the k and v caches being marked as buffers. Since buffers are considered static by torch.compile, we don't need to annotate them as `mark_static_address`. This is the pointer - https://github.com/pytorch-labs/gpt-fast/blob/main/model.py#L73


@ArthurZucker @gante Would you be willing to carry out a refactor similar to gpt-fast by making these caches an instance of nn-module? This will improve compatibility with torch.compile.

Reply:
Yeah! TBH as long as we don't see performance reduction, this should be alright (AKA just inheriting from nn.Module + marking as buffers). 
You can also potentially add it here!

Reply:
@anijain2305 sounds great! 

Will you open a PR for it? :D

Reply:
https://github.com/huggingface/transformers/pull/32159 we actually needed the deep copy so went ahead and added that for now, testing is required on my end